x-pack/plugins/security_solution/public/common/components/event_details/event_details.tsx

```diff
@@ -51,12 +49,13 @@ export const EventDetails = React.memo<Props>(
     data,
     id,
     view,
-    onEventToggled,
     onUpdateColumns,
     onViewSelected,
     timelineId,
     toggleColumn,
   }) => {
+    const handleTabClick = useCallback((e) => onViewSelected(e.id as View), [onViewSelected]);
```

Comment:
We could define an enum for eventsView, 
```
enum EventsViewType {
  'tableView' = 'table-view',
  'jsonView' = 'json-view',
}
```


and replace it with line 67 and 82 to avoid `as View` here.

```export type View = EventsViewType.tableView | EventsViewType.jsonView;``` on line 24 

x-pack/plugins/security_solution/public/common/components/event_details/stateful_event_details.tsx line 29
```
const [view, setView] = useState<View>(EventsViewType.tableView);
```